src/Microsoft.AspNetCore.Blazor.Server/MonoDebugProxy/MonoDebugProxyAppBuilderExtensions.cs

```diff
@@ -0,0 +1,178 @@
+                // when you press the debugger hotkey, include it in the querystring passed
+                // here, then remove it once the debugger connects.
+                await context.Response.WriteAsync($@"
+                    <h1>Multiple matching tabs are open</h1>
+                    <p>
+                        There is more than one browser tab at <code>{targetTabUrl}</code>.
+                        Close the ones you do not wish to debug, then refresh this page.
+                    </p>");
+                return;
+            }
+
+            // Now we know uniquely which tab to debug, construct the URL to the debug
+            // page and redirect there
+            var tabToDebug = matchingTabs.Single();
+            var underlyingV8Endpoint = tabToDebug.WebSocketDebuggerUrl;
+            var proxyEndpoint = $"{request.Host}{request.PathBase}/_framework/debug/ws-proxy?browser={WebUtility.UrlEncode(underlyingV8Endpoint)}";
+            var devToolsUrlAbsolute = new Uri(debuggerHost + tabToDebug.DevtoolsFrontendUrl);
+            var devToolsUrlWithProxy = $"{devToolsUrlAbsolute.Scheme}://{devToolsUrlAbsolute.Authority}{devToolsUrlAbsolute.AbsolutePath}?ws={proxyEndpoint}";
+            context.Response.Redirect(devToolsUrlWithProxy);
+        }
+
+        private static string GetLaunchChromeInstructions(string appRootUrl)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return $@"<p>Close this browser, then press Win+R and enter the following:</p>
+                          <p><strong><code>""%programfiles(x86)%\Google\Chrome\Application\chrome.exe"" --remote-debugging-port=9222 {appRootUrl}</code></strong></p>";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return $@"<p>Close this browser, then in a terminal window execute the following:</p>
+                          <p><strong><code>google-chrome --remote-debugging-port=9222 {appRootUrl}</code></strong></p>";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return $@"<p>Close this browser, then in a terminal window execute the following:</p>
+                          <p><strong><code>open /Applications/Google\ Chrome.app --remote-debugging-port=9222 {appRootUrl}</code></strong></p>";
+            }
+            else
+            {
+                throw new InvalidOperationException("Unknown OS platform");
+            }
+        }
+
+        private static async Task<IEnumerable<BrowserTab>> GetOpenedBrowserTabs(string debuggerHost)
+        {
+            using (var httpClient = new HttpClient { Timeout = TimeSpan.FromSeconds(5) })
+            {
+                var jsonResponse = await httpClient.GetStringAsync($"{debuggerHost}/json");
+                return JsonConvert.DeserializeObject<BrowserTab[]>(jsonResponse);
```

Comment:
This should check the status code and then use `ReadAsAsync` https://www.fuget.org/packages/System.Net.Http.Formatting.Extension/5.2.3/lib/net/System.Net.Http.Formatting.dll/System.Net.Http/HttpContentExtensions#M%3ASystem.Net.Http.HttpContentExtensions.ReadAsAsync%60%601%28System.Net.Http.HttpContent%29

This package is part of ASP.NET Core as of 2.0

Reply:
Totally happy to change this later, but I'm not sure where the `ReadAsAsync` extension method really is.

The package you linked appears not to be owned by Microsoft: https://www.nuget.org/packages/System.Net.Http.Formatting.Extension/5.2.3. The most likely standard distribution package I can find is `System.Net.Http.Formatting`, but that doesn't appear to be in `netstandard2.0`.

Since I think the behaviour will be equivalent, I've left it using `JsonConvert`. But if you know of a `netstandard2.0` way of using `ReadAsAsync` please let me know!